### ISSUE TITLE
Using different optimizers on a trained model leads to poor test accuracy.

### ISSUE THREAD

**Issue by ankit1997** [tensorflow/tensorflow#19007]
I have trained a convolutional neural network model using `AdamOptimizer`. The test accuracy is pretty good. This was expected since the validation loss was decreasing while training.

Now I restored the model weights using `saver.restore()` method, but this time I built the model with `RMSPropOptimizer` just for experimenting. Please note that I **DID NOT RETRAIN THE MODEL**. To my surprise, the test accuracy was horrible with the new optimizer. The accuracy becomes good again when I use the previously used `AdamOptimizer`.

This is peculiar in the sense that the train_op should not affect the test data results.

Below is the sample code I used for calculating the test accuracy. Please note I did not run the `train_op` here.

```
accuracies = []

# Data generator
for batchx, batchy in dataLoader.next_validation():
    # Run the graph.
    pred = sess.run(model['prediction'], 
                        feed_dict={
                            model['sensor_data']: batchx,
                            model['label']: batchy
                        })
        
    accuracies.append(np.count_nonzero(pred == label) / pred.shape[0] * 100)

accuracies = np.array(accuracies)
print("Average Validation set accuracy: {} %".format(accuracies.mean()))
```

Am I missing something here? Or is this a possible bug?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by ankit1997** [comment]
Have I written custom code: YES
OS Platform and Distribution: Ubuntu 16.04
TensorFlow installed from pip3
TensorFlow version : 1.6.0
Bazel version: NA
CUDA/cuDNN version: Cuda Toolkit 9.0, cuDNN v7.0
GPU model and memory: GeForce GTX 860M (4GB)

Exact command to reproduce: The code is quite large :(

**Comment by tatatodd** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
